packages/core/bootstrap/src/lib/external-adapter/requester.ts

```diff
@@ -70,7 +71,13 @@ export class Requester {
         if (n === 1) {
           const message = `Could not retrieve valid data: ${JSON.stringify(response.data)}`
           const cause = response.data.error || 'customError'
-          throw new AdapterError({ message, cause, url })
+          throw new AdapterError({
+            statusCode: 200,
+            providerStatusCode: response.data.error?.code || response.status,
```

Comment:
If `response.data.error?.code` is `0`, do we want to override with `response.status`? I'd suggest using `??` in place of `||` for number overrides